core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -60,6 +60,10 @@ private[spark] class TaskSetManager(
   val SPECULATION_QUANTILE = conf.getDouble("spark.speculation.quantile", 0.75)
   val SPECULATION_MULTIPLIER = conf.getDouble("spark.speculation.multiplier", 1.5)
 
+  // Threshold for task size in bytes (default is 100KB)
+  val TASK_SIZE_THRESHOLD_TO_WARN = conf.getInt("spark.task.size.warnThreshold",
```

Comment:
Are you trying to make this configuration an internal one, or you want to expose to the user? If the latter, I think we should add this to docs.

Also it would be better to make this configuration a `ConfigEntry` and add to `o.a.s.internal.config`.